maas/client/viscera/pods.py

```diff
@@ -0,0 +1,240 @@
+        return await self._handler.refresh(id=self.id)
+
+    async def parameters(self):
+        """Get the power parameters for the `Pod`."""
+        return await self._handler.parameters(id=self.id)
+
+    async def compose(
+            self, *, cores: int=None, memory: int=None,
+            cpu_speed: int=None, architecture: str=None,
+            storage: typing.Sequence[str]=None, hostname: str=None,
+            domain: typing.Union[int, str]=None,
+            zone: typing.Union[int, str, Zone]=None):
+        """Compose a machine from `Pod`.
+
+        All fields below are optional:
+
+        :param cores: Minimum number of CPU cores.
+        :type cores: `int`
+        :param memory: Minimum amount of memory (MiB).
+        :type memory: `int`
+        :param cpu_speed: Minimum amount of CPU speed (MHz).
+        :type cpu_speed: `int`
+        :param architecture: Architecture for the machine. Must be an
+            architecture that the pod supports.
+        :type architecture: `str`
+        :param storage: A list of storage constraint identifiers, in the form:
+            <label>:<size>(<tag>[,<tag>[,...])][,<label>:...]
+        :type storage: `str`
+        :param hostname: Hostname for the newly composed machine.
+        :type hostname: `str`
+        :param domain: The domain ID for the machine (optional).
+        :type domain: `int` or `str`
+        :param zone: The zone ID for the machine (optional).
+        :type zone: `int` or 'str' or `Zone`
+        :returns: The created Machine
+        :rtype: `Machine`
+        """
+        params = remove_None({
+            'cores': str(cores) if cores else None,
+            'memory': str(memory) if memory else None,
+            'cpu_speed': str(cpu_speed) if cpu_speed else None,
+            'architecture': architecture,
+            'storage': storage,
+            'hostname': hostname,
+            'domain': str(domain) if domain else None,
+        })
+        if zone is not None:
+            if isinstance(zone, Zone):
+                params["zone"] = str(zone.id)
+            elif isinstance(zone, int):
+                params["zone"] = str(zone)
+            elif isinstance(zone, str):
+                params["zone"] = zone
+            else:
+                raise TypeError(
+                    "zone must be an int, str or Zone, not %s" %
+                    type(zone).__name__)
+        return await self._handler.compose(**params, id=self.id)
+
+    async def add_tag(self, tag: str):
+        """Add tag to `Pod`.
+
+        :param tag: The tag being added.
+        :type tag: `str`
+        """
+        params = {'tag': tag}
+        return await self._handler.add_tag(**params, id=self.id)
+
+    async def remove_tag(self, tag: str):
+        """Remove tag from `Pod`.
+
+        :param tag: The tag being removed.
+        :type tag: `str`
+        """
+        params = {'tag': tag}
+        return await self._handler.remove_tag(**params, id=self.id)
```

Comment:
I am not a fan of the `add_tag` and `remove_tag` in the API. The goal of the library is to make it easier for people to work with MAAS without having to know the API.

Look at how this is handled on a machine inside the `save`. This is how it should be handled.